digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -624,12 +614,10 @@ trait LaneOperations {
   /**
     * Saves new linear assets from UI. Used by Digiroad2Api /linearassets POST endpoint.
     */
-  def create(newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String, vvhTimeStamp: Long = vvhClient.roadLinkData.createVVHTimeStamp()): Seq[Long] = {
-
-    if (newIncomeLane.isEmpty || linkIds.isEmpty)
-      return Seq()
-
-    withDynTransaction {
+  def create(newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String, vvhTimeStamp: Long = vvhClient.roadLinkData.createVVHTimeStamp(), newTransaction: Boolean = true): Seq[Long] = {
+    def createProcess(): Seq[Long] = {
+      if (newIncomeLane.isEmpty || linkIds.isEmpty)
+        return Seq()
```

Comment:
Maybe we can do this validation before if (newTransaction)

Reply:
Done